### PR TITLE
Improve AsyncState README

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes https://github.com/dotnet/extensions/issues/4611

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4764)

### CODE REVIEW

**Comment by sebastienros** [comment]
@KalleOlaviNiemitalo feedback is welcome

File: src/Libraries/Microsoft.Extensions.AsyncState/README.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # Microsoft.Extensions.AsyncState
 
-This provides the ability to store and retrieve state objects that flow with the current asynchronous context.
+This provides the ability to store and retrieve objects that flow with the current asynchronous context.
+
+It has a few advantages over using the [`AsyncLocal<T>`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.asynclocal-1) class directly:
+- By abstracting the way the ambient data is stored we can use more optimized implementations, for instance when using ASP.NET, without exposing these components.
```
[the remark this answers is not in the document]
Comment (sebastienros) [reply src/Libraries/Microsoft.Extensions.AsyncState/README.md]: Fixed, thanks

File: src/Libraries/Microsoft.Extensions.AsyncState/README.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 # Microsoft.Extensions.AsyncState
 
-This provides the ability to store and retrieve state objects that flow with the current asynchronous context.
+This provides the ability to store and retrieve objects that flow with the current asynchronous context.
+
+It has a few advantages over using the [`AsyncLocal<T>`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.asynclocal-1) class directly:
```
Comment (RussKie) [inline src/Libraries/Microsoft.Extensions.AsyncState/README.md:5]:
```suggestion
It has a few advantages over using the [`AsyncLocal<T>`](https://learn.microsoft.com/dotnet/api/system.threading.asynclocal-1) class directly:
```

Comment (sebastienros) [reply src/Libraries/Microsoft.Extensions.AsyncState/README.md:5]: I knew about it, shame on me

**Review by geeknoid — APPROVED** [review]

**Pull request merged**